services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/AbsSyncViewModelTest.java

```diff
@@ -111,4 +120,117 @@ public void checkIfSyncAttachmentState_isAvailable() {
         assertEquals(SyncAttachmentState.REDUCED_SYNC, absSyncViewModel.getCurrentSyncAttachmentState());
         assertNotNull(absSyncViewModel.getSyncAttachmentState().getValue());
     }
+
+    @Test
+    public void testNullAppName() {
+        absSyncViewModel.setAppName(null);
+        assertNull(absSyncViewModel.getAppName());
+    }
+
+    @Test
+    public void testNullServerUrl() {
+        absSyncViewModel.setServerUrl(null);
+        assertNull(absSyncViewModel.getServerUrl().getValue());
+    }
+
+    @Test
+    public void testNullUsername() {
+        absSyncViewModel.setUsername(null);
+        assertNull(absSyncViewModel.getUsername().getValue());
+    }
+
+    @Test
+    public void testIsFirstLaunchFlagTrue() {
+        absSyncViewModel.setIsFirstLaunch(true);
+        assertTrue(absSyncViewModel.checkIsFirstLaunch().getValue());
+    }
+
+    @Test
+    public void testIsFirstLaunchFlagFalse() {
+        absSyncViewModel.setIsFirstLaunch(false);
+        assertFalse(absSyncViewModel.checkIsFirstLaunch().getValue());
+    }
+
+    @Test
+    public void testServerUrlExtremeValue() {
+        String longUrl = "http://" + String.join("", Collections.nCopies(100000, "a")) + ".com";
+        absSyncViewModel.setServerUrl(longUrl);
+        assertEquals(longUrl, absSyncViewModel.getServerUrl().getValue());
+    }
+
+    @Test
+    public void testUserState() {
+        for (UserState state : UserState.values()) {
+            absSyncViewModel.setCurrentUserState(state);
+        }
+        assertEquals(UserState.values()[UserState.values().length - 1], absSyncViewModel.getCurrentUserState().getValue());
+    }
+
+    @Test
+    public void testLastSyncTimeObserver() throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final Observer<Long> observer = new Observer<Long>() {
+            @Override
+            public void onChanged(Long value) {
+                assertEquals(1000L, (long) value);
+                latch.countDown();
+            }
+        };
+
+        absSyncViewModel.getLastSyncTime().observeForever(observer);
+        absSyncViewModel.setLastSyncTime(1000L);
+        latch.await(2, TimeUnit.SECONDS);
+        absSyncViewModel.getLastSyncTime().removeObserver(observer);
+    }
+
+    @Test
```

Comment:
Great test, but what is the main value of this test, since the getLastSyncTime is accessed from the UI thread. From what I understand, the lastSyncTime represents the last time the data was synchronised with the server. @wbrunette  any thoughts?

Reply:
> Great test, but what is the main value of this test, since the getLastSyncTime is accessed from the UI thread. From what I understand, the lastSyncTime represents the last time the data was synchronised with the server. @wbrunette  any thoughts?

@wbrunette @lamouresparus Hi, I trust you had a restful weekend. I'm still waiting on how to proceed with this?